src/main/java/tr/com/srdc/cda2fhir/transform/CCDTransformerImpl.java

```diff
@@ -197,23 +214,29 @@ public Bundle transformDocument(ClinicalDocument cda) {
         }
 
         // init the global ccd bundle via a call to resource transformer, which handles cda header data (in fact, all except the sections)
-        Bundle ccdBundle = resTransformer.tClinicalDocument2Composition(ccd);
+        Bundle ccdBundle = resTransformer.tClinicalDocument2Bundle(ccd, includeComposition);
+        
         // the first bundle entry is always the composition
-        Composition ccdComposition = (Composition)ccdBundle.getEntry().get(0).getResource();
+        Composition ccdComposition = includeComposition ? (Composition)ccdBundle.getEntry().get(0).getResource() : null;
+        
         // init the patient id reference if it is not given externally. the patient is always the 2nd bundle entry
-        if (patientRef == null)
-            patientRef = new Reference(ccdBundle.getEntry().get(1).getResource().getId());
-        else // Correct the subject at composition with given patient reference.
+        if (patientRef == null) {
+            patientRef = new Reference(ccdBundle.getEntry().get(1).getResource().getId()); // TO DO: Do not depend on the order
```

Comment:
This is going to break if the composition is removed. Maybe:
```
patientRef = includeCompositon ? 
new Reference(ccdBundle.getEntry().get(1).getResource().getId()) :
new Reference(ccdBundle.getEntry().get(0).getResource().getId());
```

Reply:
This will also be changed once order is removed.

Reply:
Allright, i made it independent for order.